apps/web/src/pages/workflow/views/editor/components/config-panel/components/param-input/index.tsx

```diff
@@ -3,6 +3,187 @@
  *
  * Note: use in TriggerNode, CodeNode
  */
-const ParamInput = () => {};
+
+import { useLayoutEffect, useMemo } from 'react';
+import {
+    Select,
+    Button,
+    IconButton,
+    FormControl,
+    MenuItem,
+    TextField,
+    Switch,
+    InputLabel,
+    type SelectProps,
+    type TextFieldProps,
+} from '@mui/material';
+import { useDynamicList, useControllableValue } from 'ahooks';
+import { useI18n } from '@milesight/shared/src/hooks';
+import { DeleteOutlineIcon, AddIcon } from '@milesight/shared/src/components';
+import './style.less';
+
+export type ParamInputValueType = {
+    arg: string;
+    type: string;
+    isCheck?: boolean;
+};
+
+export interface ParamInputProps {
+    required?: boolean;
+    disabled?: boolean;
+    defaultValue?: ParamInputValueType[];
+    showSwitch?: boolean;
+    typeSelectProps?: SelectProps;
+    nameInputProps?: TextFieldProps;
+}
+
+// TODO: is need?
+const MAX_VALUE_LENGTH = 10;
+
+const ParamInput: React.FC<ParamInputProps> = ({
+    required,
+    disabled,
+    defaultValue,
+    showSwitch,
+    typeSelectProps,
+    nameInputProps,
+    ...props
+}) => {
+    const { getIntlText } = useI18n();
+    const [innerValue, setInnerValue] = useControllableValue<ParamInputValueType[]>(props, {
+        defaultValue: defaultValue || [],
+    });
+    const { list, remove, getKey, insert, replace } = useDynamicList<ParamInputValueType>(
+        innerValue || [],
+    );
+
+    const DEFAULT_EMPTY_VALUE: ParamInputValueType = useMemo(() => {
+        if (showSwitch) {
+            return {
+                arg: '',
+                type: '',
+                isCheck: false,
+            };
+        }
+        return {
+            arg: '',
+            type: '',
+        };
+    }, [showSwitch]);
```

Comment:
`DEFAULT_EMPTY_VALUE` should be a constant, using `useMemo` will increase the performance loss, and `isCheck` is an optional parameter, `undefined` means `false`, there is no need to configure it here.